packages/authentication/src/__tests__/acceptance/jwt-auth-extension.acceptance.ts

```diff
@@ -0,0 +1,406 @@
+        else return 'userProfile is undefined';
+      }
+    }
+
+    app.controller(InfoController);
+
+    const the_token: string = (await whenIMakeRequestTo(server)
+      .get('/login')
+      .expect(200)).text;
+
+    expect(the_token !== null).to.equal(true);
+    expect(typeof the_token === 'string').to.equal(true);
+
+    // not specifying an `Authorization` header value
+    await whenIMakeRequestTo(server)
+      .get('/whoAmI')
+      .set('Authorization', 'NotB3ar3r ' + the_token)
+      .expect(401);
+  });
+
+  it('returns error due to expired token', async () => {
+    class InfoController {
+      constructor() {}
+
+      @get('/whoAmI')
+      @authenticate('jwt')
+      whoAmI(
+        @inject(AuthenticationBindings.CURRENT_USER) userProfile: UserProfile,
+      ) {
+        if (userProfile) {
+          if (userProfile.email) return userProfile.email;
+          else return 'userProfile email is undefined';
+        } //if
+        else return 'userProfile is undefined';
+      }
+    }
+
+    app.controller(InfoController);
+
+    const expiredToken =
+      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjEiLCJlbWFpbCI6ImpvZUBleGFtcGxlLmNvbSIsIm5hbWUiOiJqb2Ugam9lbWFuIiwiaWF0IjoxNTU1ODY3NDAzLCJleHAiOjE1NTU4Njc0NjN9.QKmO5qDC8Yg-aK3EedLRsXczL7VQDDnWtA-cpyqszqM';
+
+    await whenIMakeRequestTo(server)
+      .get('/whoAmI')
+      .set('Authorization', 'Bearer ' + expiredToken)
+      .expect(401);
+  });
+
+  it('returns error due to invalid token #1', async () => {
+    class InfoController {
+      constructor() {}
+
+      @get('/whoAmI')
+      @authenticate('jwt')
+      whoAmI(
+        @inject(AuthenticationBindings.CURRENT_USER) userProfile: UserProfile,
+      ) {
+        if (userProfile) {
+          if (userProfile.email) return userProfile.email;
+          else return 'userProfile email is undefined';
+        } //if
+        else return 'userProfile is undefined';
+      }
+    }
+
+    app.controller(InfoController);
+
+    const invalidToken = 'aaa.bbb.ccc';
+
+    await whenIMakeRequestTo(server)
+      .get('/whoAmI')
+      .set('Authorization', 'Bearer ' + invalidToken)
+      .expect(401);
+  });
+
+  it('returns error due to invalid token #2', async () => {
+    class InfoController {
+      constructor() {}
+
+      @get('/whoAmI')
+      @authenticate('jwt')
+      whoAmI(
+        @inject(AuthenticationBindings.CURRENT_USER) userProfile: UserProfile,
+      ) {
+        if (userProfile) {
+          if (userProfile.email) return userProfile.email;
+          else return 'userProfile email is undefined';
+        } //if
+        else return 'userProfile is undefined';
+      }
+    }
+
+    app.controller(InfoController);
+
+    const invalidToken = 'aaa.bbb.ccc.ddd';
+
+    await whenIMakeRequestTo(server)
+      .get('/whoAmI')
+      .set('Authorization', 'Bearer ' + invalidToken)
+      .expect(401);
+  });
+
+  it('create a json web token throws error for missing email', async () => {
```

Comment:
Nitpick (same for the test below): `create` -> `creates`, 
`and throws`?

Reply:
We use singular form for the verb to describe a test.

Reply:
@raymondfeng , are you agreeing with me or @nora?

Reply:
Yes, I agree with @nabdelgadir to use `creates`.